### PR TITLE
Add utils::locale_to_utf8()

### PR DESCRIPTION
**Pull request by Minoru**
This is a little detour from #1344, but kind of productive as well. Added some tests, plus a little helper function to save on typing in a few places.

Reviews are welcome! If no issues are raised, I'll merge in three days.

### CODE REVIEW

**Review by dennisschagt — APPROVED** [review]
Nice changes, LGTM

File: test/utils.cpp [hunk 1]
```diff
@@ -1719,3 +1721,185 @@ TEST_CASE("mkdir_parents() doesn't care if the path ends in a slash or not",
...
+				REQUIRE(utils::convert_text(input, tocode, fromcode) == input);
+			}
+		}
+	}
+}
+
+TEST_CASE("convert_text() returns empty string if conversion is impossible",
+	"[utils]")
+{
+	// "Hello", in Russian, encoded in UTF-8.
```
[the remark this answers is not in the document]
Comment (Minoru) [reply test/utils.cpp]: Indeed! Fixed in https://github.com/newsboat/newsboat/pull/1443/commits/3709ed600f4ac54eb68c9d75cd1bb068b9e52071.

File: src/utils.cpp [hunk 2]
```diff
@@ -320,8 +320,21 @@ std::string utils::convert_text(const std::string& text,
...
-	iconv_t cd = ::iconv_open(
-			translit(tocode, fromcode).c_str(), fromcode.c_str());
+	const auto tocode_translit = translit(tocode, fromcode);
+
+	// Illegal and incomplete multi-byte sequences will be replaced by this
+	// placeholder. By default, we use an ASCII value for "question mark".
+	std::string question_mark("\x3f");
+	// This `if` prevens the function to recurse indefinitely.
+	if (text != question_mark && fromcode != "ASCII") {
+		question_mark = utils::convert_text("\x3f", tocode_translit.c_str(), "ASCII");
```
[the remark this answers is not in the document]
Comment (Minoru) [reply src/utils.cpp]: Fixed: https://github.com/newsboat/newsboat/compare/7f6dd5890afc484d9f17e7d29d680b2b689df7fd..3709ed600f4ac54eb68c9d75cd1bb068b9e52071

**Comment by dennisschagt** [comment]
Looks good, thanks for the changes

**Comment by Minoru** [comment]
Thank you for the review!

**Pull request merged**